spec/models/user_notification_spec.rb

```diff
@@ -0,0 +1,44 @@
+# == Schema Information
+#
+# Table name: user_notification
+#
+#  course_update :boolean          default("false")
+#  created_at    :datetime         not null
+#  id            :bigint           not null, primary key
+#  provider_code :string           not null
+#  updated_at    :datetime         not null
+#  user_id       :integer          not null
+#
+describe UserNotification, type: :model do
+  describe "associations" do
+    let(:organisation) { create(:organisation, providers: [provider]) }
+    let(:user) { create(:user, organisations: [organisation]) }
+    let(:provider) { create(:provider) }
+
+    subject { UserNotification.new(user_id: user.id, provider_code: provider.provider_code) }
+
+    it { should belong_to(:provider) }
+    it { should belong_to(:user) }
+  end
+
+  describe "scopes" do
+    describe "#for_provider" do
+      let(:user) { create(:user, organisations: [organisation]) }
+      let(:organisation) { create(:organisation, providers: [provider_one, provider_two]) }
+      let(:provider_one) { create(:provider) }
+      let(:provider_two) { create(:provider) }
+
+      before do
+        UserNotification.new(user_id: user.id, provider_code: provider_one.provider_code).save!
+        UserNotification.new(user_id: user.id, provider_code: provider_two.provider_code).save!
+      end
+
+      it "Returns only user notifications the first provider" do
+        notifications = UserNotification.for_provider(provider_one.provider_code).all
```

Comment:
```suggestion
        notifications = described_class.for_provider(provider_one.provider_code)
```
maybe use `described_class`

I don't think there is a need for `.all`

I'm not sure  of `for_provider`

```ruby 
UserNotification.find_by(provider_code: provider_one.provider_code)
```

In fact do we actually need it....
```ruby
provider.user_notifications # give lists of users...
course.provider.user_notifications # give lists of users...
course.accrediting_provider.user_notifications # give lists of users that belongs to accrediting_provider...
```

`for_provider` I leave that for your consideration

Reply:
Described class could work for sure - will give it a poke

Though I do think you're right here, the trello task implied that methods would be needed, but didnt specify. I think I could scrap this and do the follow up card and see if any methods are required after doing that